syntax = "proto3";

package signal;


message Complex {
    double real = 1;
    double imag = 2;
}

message respcoeffs{
    repeated double coe = 1;
}

message SignalGrpc {
    repeated Complex values = 1;
}

message CWTRequest {
    SignalGrpc signal = 1;   
    double start = 2;
    double end = 3;
    int32 numScales = 4; 
}


message CWTResponse  {
    repeated respcoeffs coeffs = 1;
}


service SignalService {
    rpc ComputeCWT (CWTRequest) returns (CWTResponse);
}